arcgis-ios-sdk-samples/Geometry/Geodesic operations/GeodesicOperationsViewController.swift

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class GeodesicOperationsViewController: UIViewController, AGSGeoViewTouchDelegate {
+
+    @IBOutlet private var mapView: AGSMapView!
+    
+    private let destinationGraphic: AGSGraphic
+    private let pathGraphic: AGSGraphic
+    
+    private let graphicsOverlay = AGSGraphicsOverlay()
+    private let measurementFormatter = MeasurementFormatter()
+    private let JFKAirportLocation = AGSPoint(x: -73.7781, y: 40.6413, spatialReference: AGSSpatialReference.wgs84())
+    
+    /// Add graphics representing origin, destination, and path to a graphics overlay.
+    required init?(coder aDecoder: NSCoder) {
+        // Create graphic symbols.
+        let locationMarker = AGSSimpleMarkerSymbol(style: .cross, color: .blue, size: 10)
+        let pathSymbol = AGSSimpleLineSymbol(style: .dash, color: .blue, width: 5)
+        
+        // Create an origin graphic.
+        let originGraphic = AGSGraphic(geometry: JFKAirportLocation, symbol: locationMarker, attributes: nil)
+        
+        // Create a destination graphic.
+        destinationGraphic = AGSGraphic(geometry: nil, symbol: locationMarker, attributes: nil)
+        
+        // Create a graphic to represent geodesic path between origin and destination.
+        pathGraphic = AGSGraphic(geometry: nil, symbol: pathSymbol, attributes: nil)
+        
+        super.init(coder: aDecoder)
+        
+        // Add graphics to the graphics overlay.
+        graphicsOverlay.graphics.add(originGraphic)
```

Comment:
Just curious: is there a reason these come after the `super.init()`?

Also, they could be consolidated into a single line with something like:

``` Swift
graphicsOverlay.graphics.addObjects(from: [originGraphic, destinationGraphic, pathGraphic])

super.init(coder: aDecoder)
```

Reply:
My idea was to initialize `destinationGraphic` and `pathGraphic` before calling `super.init()` and put rest of the code that does not involve initialization of immutable and non-optional properties after `super.init()`. @philium any suggestions if `graphicsOverlay.graphics.addObjects(from: [originGraphic, destinationGraphic, pathGraphic])` should be after `super.init()` from best practices point of view or does it not matter?

Reply:
@gdhakal I would consider setting the overlay's `graphics` as part of the initialization of `graphicsOverlay`. To put it another way, `graphicsOverlay` (and therefore `self`) isn't fully initialized until the graphics have been added to the overlay. From that perspective, it would make sense to do that before the call to `super.init`.

That being said, the way you did it isn't wrong. The class behaves exactly the same either way.